packages/modules/cra-template-web-viewer/template/src/App.tsx

```diff
@@ -44,121 +35,63 @@ const App: React.FC = () => {
-    await IModelApp.authorizationClient?.signOut();
-    setIsAuthorized(false);
-  };
+  }
 
-  const onIModelAppInit = () => {
-    setIsAuthorized(IModelApp.authorizationClient?.isAuthorized || false);
-    IModelApp.authorizationClient?.onUserStateChanged.addListener(() => {
-      setIsAuthorized(
-        (IModelApp.authorizationClient?.hasSignedIn &&
-          IModelApp.authorizationClient?.isAuthorized) ||
-          false
+  if (urlParams.has("iModelId")) {
+    sessionStorage.setItem(
+      "IMJS_IMODEL_ID",
+      urlParams.get("iModelId") as string
+    );
+  } else {
+    if (process.env.IMJS_IMODEL_ID) {
+      sessionStorage.setItem("IMJS_IMODEL_ID", process.env.IMJS_IMODEL_ID);
+    } else {
+      throw new Error(
+        "Please add a valid iModel ID in the .env file and restart the application or add it to the iModelId query parameter in the url and refresh the page. See the README for more information."
       );
-    });
-  };
+    }
+  }
 
-  /** NOTE: This function will execute the "Fit View" tool after the iModel is loaded into the Viewer.
-   * This will provide an "optimal" view of the model. However, it will override any default views that are
-   * stored in the iModel. Delete this function and the prop that it is passed to if you prefer
-   * to honor default views when they are present instead (the Viewer will still apply a similar function to iModels that do not have a default view).
-   */
-  const viewConfiguration = (viewPort: ScreenViewport) => {
-    const tileTreesLoaded = () => {
-      return new Promise((resolve, reject) => {
-        const start = new Date();
-        const intvl = setInterval(() => {
-          if (viewPort.areAllTileTreesLoaded) {
-            clearInterval(intvl);
-            resolve(true);
-          }
-          const now = new Date();
-          // after 20 seconds, stop waiting and fit the view
-          if (now.getTime() - start.getTime() > 20000) {
-            reject();
-          }
-        }, 100);
-      });
-    };
+  /** Creation of a client */
+  let client: FrontendAuthorizationClient;
+  if (AuthorizationOptions.config) {
+    client = new BrowserAuthorizationClient(AuthorizationOptions.config);
+  } else if (AuthorizationOptions.oidcClient) {
+    client = AuthorizationOptions.oidcClient;
+  } else {
+    throw new Error("Please provide a valid AuthOptions file.");
+  }
 
-    tileTreesLoaded().finally(() => {
-      IModelApp.tools.run(FitViewTool.toolId, viewPort, true, false);
-      viewPort.view.setStandardRotation(StandardViewId.Iso);
-    });
-  };
+  /** Handle the callback, otherwise just signin. */
+  BrowserAuthorizationCallbackHandler.handleSigninCallback(
+    process.env.IMJS_AUTH_CLIENT_REDIRECT_URI
+  ).then(() => {
+    client.signIn(new FrontendRequestContext());
+    history.push(
+      `?contextId=${sessionStorage.getItem(
+        "IMJS_CONTEXT_ID"
+      )}&iModelId=${sessionStorage.getItem("IMJS_IMODEL_ID")}`
+    );
+  });
```

Comment:
Signing in on startup proved to be the most challenging task. Because of redirection, the client has no idea if it is signed in or not, and there is no easy method call to determine if there is an OIDC token currently in local storage. This will handle the callback, if a signin request has already been made, then attempt to signin again. With the second signin request, it will silent signing because the browser already has a token. Then the history is pushed because if it is pushed before this point, the callbackHandler is unable to get a token from the return url.

I couldn't find a more elegant solution, but this atleast works. Please give thoughts and opinions!